tests/lib/rules/consistent-return.js

```diff
@@ -42,7 +42,11 @@ ruleTester.run("consistent-return", rule, {
 
         // https://github.com/eslint/eslint/issues/7790
         { code: "class Foo { constructor() { if (true) return foo; } }", parserOptions: { ecmaVersion: 6 } },
-        { code: "var Foo = class { constructor() { if (true) return foo; } }", parserOptions: { ecmaVersion: 6 } }
+        { code: "var Foo = class { constructor() { if (true) return foo; } }", parserOptions: { ecmaVersion: 6 } },
+
+        // https://github.com/eslint/eslint/issues/11371
+        { code: "function* foo() { if (false) { return []; } yield* [1, 2, 3];}", parserOptions: { ecmaVersion: 6 } },
```

Comment:
Can we add tests for when this would error? For example, when there isn't a `yield` or `return` statement in a code path?

Reply:
Are you saying for generator functions or regular functions?

I can add an example for a regular function, a generator function would actually pass this rule. `function* foo() { var x = 5; }` would fail [require-yield](https://eslint.org/docs/rules/require-yield)

Reply:
I think this rule should still check all code paths, so something like this should still fail this rule:

```
function* foo(bar) { 
  if (bar === 'a') { 
    return []; 
  } else if (bar === 'b') {
    yield* [1, 2, 3];
  } else {
    var x = 5;
  }
}
```

Reply:
Friendly ping :)